deepchecks/tabular/dataset.py

```diff
@@ -1061,6 +1065,172 @@ def datasets_share_date(cls, *datasets: 'Dataset') -> bool:
 
         return True
 
+    @property
+    def _dataset_description(self) -> t.Tuple[
+        t.List[t.Union[str, int]],
+        pd.DataFrame,
+        pd.DataFrame
+    ]:
+        data = self.data
+        datetime_name = self.datetime_name
+        index_name = self.index_name
+        label_name = self.label_name
+        features = self.features
+        categorical_features = self.cat_features
+        numerical_features = self.numerical_features
+
+        columns = features[::]
+
+        if label_name is not None:
+            columns = [*columns, label_name]
+
+        is_index_name_str = isinstance(index_name, str)
+        is_index_name_int = isinstance(index_name, int)
+        is_multiindex = isinstance(data.index, pd.MultiIndex)
+
+        unamed_index_levels = {
+            level
+            for level, name in enumerate(data.index.names)
+            if name is None
+        }
+
+        if index_name in data.columns:
+            columns = [index_name, *columns]
+
+        elif (is_index_name_str or is_index_name_int) and index_name in data.index.names:
+            data = data.reset_index()
+            columns = [index_name, *columns]
+
+        elif is_index_name_int and is_multiindex and index_name in unamed_index_levels:
+            data = data.reset_index()
+            index_name = f'level_{index_name}'
+            columns = [index_name, *columns]
+
+        else:
+            data = data.reset_index()
+            index_name = 'index'
+            columns = [index_name, *columns]
+
+        is_datetime_name_str = isinstance(datetime_name, str)
+        is_datetime_name_int = isinstance(index_name, int)
+
+        if datetime_name in columns:
```

Comment:
what's the difference between `datetime_name in columns` and `datetime_name in data.columns`?

Reply:
-- `datetime_name in columns` --

user passed `datetime_name=<something> and set_datetime_from_dataframe_index=False`
therefore datetime column could be within `columns`

-- `datetime_name in data.columns` --

user passed `datetime_name=<something> and set_datetime_from_dataframe_index=True`
datetime column is an index or one of the levels of the `multiindex` but there are possibility that previously `data.reset_index` was called therefore datetime column now could be within `data.column`

Reply:
why not just always do `reset_index()` right on the start? anyway you later filter it by columns

Reply:
I rewrote that method, this is not actual anymore